networkoperator/v1/001-egressrouter-crd.yaml

```diff
@@ -0,0 +1,231 @@
+    listKind: EgressRouterList
+    plural: egressrouters
+    singular: egressrouter
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .status.conditions[*].type
+      name: Condition
+      type: string
+    - jsonPath: .status.conditions[*].status
+      name: Status
+      type: string
+    name: v1
+    schema:
+      openAPIV3Schema:
+        description: "EgressRouter is a feature allowing the user to define an egress
+          router that acts as a bridge between pods and external systems. The egress
+          router runs a service that redirects egress traffic originating from a pod
+          or a group of pods to a remote external system or multiple destinations
+          as per configuration. \n It is consumed by the cluster-network-operator.
+          More specifically, given an EgressRouter CR with <name>, the CNO will create
+          and manage: - A service called <name> - An egress pod called <name> - A
+          NAD called <name> \n EgressRouter is a single egressrouter pod configuration
+          object."
+        type: object
+        required:
+        - spec
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: Specification of the desired egress router.
+            type: object
+            required:
+            - addresses
+            - mode
+            - networkInterface
+            properties:
+              addresses:
+                description: List of IP addresses to configure on the pod's secondary
+                  interface.
+                type: array
+                minItems: 1
+                items:
+                  description: EgressRouterAddress contains a pair of IP CIDR and
+                    gateway to be configured on the router's interface
+                  type: object
+                  required:
+                  - ip
+                  properties:
+                    gateway:
+                      description: IP address of the next-hop gateway, if it cannot
+                        be automatically determined. Can be IPv4 or IPv6.
+                      type: string
+                    ip:
```

Comment:
there is a format for IPs. Otherwise ppl can put garbage into this field.

Reply:
The problem is the IPv6 format. It is hard to check its pattern - the blob is too big. Hence we decided to move the check to the controller.

Reply:
I am talking about https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/types_jsonschema.go#L34.

Reply:
wow, wasn't aware I could do that. thanks Stefan!! will update that.